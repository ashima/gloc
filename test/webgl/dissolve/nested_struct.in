struct s { float q; };

struct h { s g; s b; } f(void) { return h(s(1.0),s(0.0)); }

invariant varying struct n { h d; } vvv;
