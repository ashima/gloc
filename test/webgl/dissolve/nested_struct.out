{
  "glo": [ 1, 0, 0 ],
  "target": [ "webgl", [ 1, 0, 0 ] ],
  "meta": { "copyright": [ 2012, [ "", "" ] ] },
  "units": [
    { "outsym": [ "s" ], "source": "#line 1 GLOC_0\nstruct s { float q; };" },
    {
      "insym": [ "s" ],
      "outsym": [ "h" ],
      "source": "#line 3 GLOC_0\nstruct h { s g; s b; };"
    },
    {
      "insym": [ "h", "s" ],
      "outsym": [ "f" ],
      "source":
        "#line 3 GLOC_0\nh                      f(void) { return h(s(1.0),s(0.0)); }"
    },
    {
      "insym": [ "h" ],
      "outsym": [ "n" ],
      "source": "#line 5 GLOC_0\n                  struct n { h d; };"
    },
    {
      "vary": [ "vvv" ],
      "insym": [ "n" ],
      "outsym": [ "vvv" ],
      "source": "#line 5 GLOC_0\ninvariant varying n                 vvv;"
    }
  ],
  "linkmap": { "0": "#n=0" }
}
